main.go

```diff
@@ -8,141 +8,131 @@ import (
 	"net/url"
 	"os"
 	"strings"
+
+	"github.com/asaskevich/govalidator"
 )
 
+func validateUrl(u string) (*url.URL, error) {
+	// Verify the target is HTTP or HTTPS.
+	if !strings.HasPrefix(u, "http://") && !strings.HasPrefix(u, "https://") {
+		return nil, fmt.Errorf("%s: must be an http(s) URL.\n", u)
+	}
+
+	// Validate the URL.
+	if !govalidator.IsURL(u) {
+		return nil, fmt.Errorf("%s: must be a valid URL.\n", u)
+	}
+
+	// Parse the target as a URL.
+	ua, err := url.Parse(u)
+	if err != nil {
+		panic(err)
+	}
+
+	return ua, nil
+}
+
 func main() {
-	if len(os.Args) != 4 {
-		fmt.Println("usage: multireq <listen addr> <target A> <target B>")
+	if len(os.Args) < 3 {
+		fmt.Fprintln(os.Stderr, "USAGE: multireq [listen-addr] [target]...")
 		os.Exit(1)
 	}
 
 	listen := os.Args[1]
 	targets := os.Args[2:]
-	if !strings.HasPrefix(targets[0], "http") {
-		fmt.Println("must specify http targets")
-		os.Exit(1)
-	}
-	if !strings.HasPrefix(targets[1], "http") {
-		fmt.Println("must specify http targets")
-		os.Exit(1)
-	}
+	urls := make([]*url.URL, len(targets))
 
-	ua, err := url.Parse(targets[0])
-	if err != nil {
-		panic(err)
+	failed := false
+	for i, v := range targets {
+		ua, err := validateUrl(v)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			failed = true
```

Comment:
True -- my intent here was to print out **all** of the malformed URLs before bailing.